File: src/com/keguoyu/minijvm/MiniJVM.java
package com.keguoyu.minijvm;

import com.keguoyu.minijvm.main.JavaVirtualMachine;
import com.keguoyu.minijvm.pojo.Args;

/**
 * @author keguoyu
 * @version 1.0
 **/
public class MiniJVM {

    public static void main(String[] args) {
        Args argv = new Args(args);
        if (!argv.isRightFmt || argv.helpFlag || argv.args == null) {
            argv.printUsage();
        } else if (argv.versionFlag) {
            System.out.println("MiniJVM Version 1.0");
        } else {
            JavaVirtualMachine jvm = new JavaVirtualMachine();
            jvm.start(argv.cpOption, argv.clazz);
        }
    }

}

File: minijvm.py
import sys

from args.arguments import Arguments
from vm.virtual_machine import VirtualMachine


def start():
    args = Arguments(sys.argv)
    if not args.right_fmt or args.help_flag:
        print("Usage: python3 minijvm.py [-classpath classpath] class")
    elif args.version_flag:
        print("MiniJVM by python version 1.0")
    else:
        VirtualMachine(args.class_path, args.class_name).start()


if __name__ == '__main__':
    start()
